code/datums/world_topic.dm

```diff
@@ -91,6 +91,12 @@
 	require_comms_key = TRUE
 
 /datum/world_topic/comms_console/Run(list/input)
+	// Reject comms messages from other servers that are not on our configured network,
+	// if this has been configured. (See CROSS_COMMS_NETWORK in comms.txt)
+	var/configured_network = CONFIG_GET(string/cross_comms_network)
+	if (configured_network && configured_network != input["network"])
+		return
+
```

Comment:
What if make `configured_network` a list of networks, so we can set bunch of nets that we want hear.
```
var/list/configured_network = params2list(CONFIG_GET(string/cross_comms_network))
	if (configured_network && configured_network & input["network"])
		return
```
where config in string "servet1;server2;server3"

Reply:
why would you have 3 servers, A/B/C, where A and C listen to B but not each other?

I guess my question is why would you not just make ABC a network as opposed to AB and BC network?

Reply:
i think about this like servers from different hosts, like citadel, tg , hippi, etc.

Reply:
do you really want that :^)

Reply:
I think its good if MRP can hear only MRP, but LRP can hear LRP and MRP

Reply:
Kinda good not gonna lie

Reply:
you can already accomplish LRP seeing MRP but not vice versa by just configuring the network on MRP but not on LRP, this results in LRP not filtering any messages for network

Reply:
> i think about this like servers from different hosts, like citadel, tg , hippi, etc. tgMRP, tgLRP, tg marines

this cannot be accomplished without exposing a comms key to other servers and with further modifications which im assuming we don't want

do you have any other reason (a specific example) for multiple networks aside from what I said above, wherein that can already be accomplished

Reply:
As i say above.
MRP can hear only MRP, but LRP can hear LRP and MRP servers.
As i see now, TG has 2 RP servers one Marine and 3 another servers.

Reply:
Dennok is correct, such a system would have been much much more flexible and expansive if it was based off of tags and tag matching.

Each server should be able to have multiple tags.

Each server should be able to specify multiple tags they will listen to events from.

Then we can do shit like a tag for each server, a tag for mrp vs lrp, a tag for event servers, (so that, say, events from event servers go out to all other servers, but receive none from any other server, including event servers).

Also each event type should be able to have tag filters. ahelp, coms console, round end notices, etc.

This is why paid code is bad, it incentives coding the system as described in some hardly thought out posting.

Reply:
> why would you have 3 servers, A/B/C, where A and C listen to B but not each other?

You shouldn't be asking why. you should be asking why not.

Reply:
this feedback would have been more helpful 9 days ago, your opinion has more weight than dennoks and would have likely led to this change being made

at this point it would be better to post on the forums in a feature request

Reply:
I shouldn't be needed to show up into every pr and do all of the idea guy work.

You should have thought of this from the start =P